### PR TITLE
Add CEL datatype

### PR DESCRIPTION
**Pull request by nturaga**

### CODE REVIEW

**Comment by martenson** [comment]
Seems fine. Would you mind adding a test for the sniffer? (example here: https://github.com/galaxyproject/galaxy/pull/2452/files#diff-d37c4977fceebe2a0c1d657c53991103R507)

**Comment by nturaga** [comment]
@martenson Thanks. I'll look now.

File: config/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -429,6 +429,7 @@
     <datatype extension="hmm3" type="galaxy.datatypes.msa:Hmmer3" display_in_upload="true" />
     <datatype extension="stockholm" type="galaxy.datatypes.msa:Stockholm_1_0" display_in_upload="True" />
     <datatype extension="xmfa" type="galaxy.datatypes.msa:MauveXmfa" display_in_upload="True" />
+    <datatype extension="CEL" type="galaxy.datatypes.binary:Cel" display_in_upload="True" />
```
Comment (nsoranzo) [inline config/datatypes_conf.xml.sample]: I think we recommend the extension to be lowercase, see also: https://travis-ci.org/galaxyproject/galaxy/jobs/168975520#L1056

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -124,6 +124,23 @@ def sniff( self, filename ):
 Binary.register_sniffable_binary_format("idat", "idat", Idat)
 
 
+class Cel( Binary ):
+    """Binary data in CEL format"""
+    file_ext = "CEL"
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]: Same as above.

File: lib/galaxy/datatypes/binary.py [hunk 3]
```diff
@@ -124,6 +124,23 @@ def sniff( self, filename ):
...
+    def sniff( self, filename ):
+        try:
+            header = open( filename, 'rb' ).read(4)
+            if header == b';\x01\x00\x00':
+                return True
+            return False
+        except:
+            return False
+
+Binary.register_sniffable_binary_format("CEL", "CEL", Cel)
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]: And again here.

**Comment by martenson** [comment]
@nturaga it seems that none of the binary datatypes have a test though, not sure if I am asking for something weird...Can you create a small sniffable binary or is it madness?

**Comment by nturaga** [comment]
None of the binary datatypes have a sniffer, from what I see too. It might not be a thing.

**Comment by martenson** [comment]
@nturaga there seems to be plenty of sniffers in `binary.py`...

File: lib/galaxy/datatypes/binary.py [hunk 4]
```diff
@@ -124,6 +124,23 @@ def sniff( self, filename ):
 Binary.register_sniffable_binary_format("idat", "idat", Idat)
 
 
+class Cel( Binary ):
+    """Binary data in CEL format"""
+    file_ext = "CEL"
+    edam_format = "format_1638"
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py:131]: Add:

```
edam_data = "data_3110"
```

**Comment by nsoranzo** [comment]
Thanks @nturaga, LGTM! I'll leave it to @martenson now.

**Comment by nturaga** [comment]
@galaxybot test this

**Comment by martenson** [comment]
@nturaga the failing test is unrelated

**Comment by martenson** [comment]
Thanks for adding the doctest @nturaga !

**Pull request merged**